manticore/core/cpu/arm.py

```diff
@@ -335,6 +335,15 @@ def _set_mode(self, new_mode):
 		self.mode = new_mode
 		self._md.mode = new_mode
 
+    def _swap_mode(self):
+        #swap from arm to thumb or back
+        assert self.mode in (CS_MODE_ARM, CS_MODE_THUMB)
+        if self.mode == CS_MODE_ARM:
+            self._set_mode(CS_MODE_THUMB)
```

Comment:
in the future (not this pr), perhaps we could use a property instead of a setter for mode